stratum/hal/lib/tdi/dpdk/dpdk_hal.h

```diff
@@ -81,7 +82,8 @@ class DpdkHal final {
                                   SwitchInterface* switch_interface,
                                   AuthPolicyChecker* auth_policy_checker,
                                   absl::Notification* ready_sync = nullptr,
-                                  absl::Notification* done_sync = nullptr)
+                                  absl::Notification* done_sync = nullptr,
+                                  const TargetOptions* target_options = nullptr)
```

Comment:
const TargetOptions* target_options =TargetOptions::default_target_options?
Since TargetOptions::default_target_options is created to be the default?

Reply:
This approach works at the expense of exposing the default target options to sections of the code that don't need to know the implementation details. I prefer passing a null pointer and supplying the defaults at the end of the line.

It also introduces tight coupling in the interface, which could cause problems in the unit tests.

I think I have an alternative approach, but I'd prefer to merge this commit before doing any additional tinkering. I believe it meets Google's "senior" code review principle:

> In general, reviewers should favor approving a CL once it is in a state where it definitely improves the overall code health of the system being worked on, even if the CL isn’t perfect.